src/cz/crcs/ectester/standalone/libs/jni/NativeKeyPairGeneratorSpi.java

```diff
@@ -306,4 +311,40 @@ public Matrixssl() {
         @Override
         native KeyPair generate(AlgorithmParameterSpec params, SecureRandom random);
     }
+
+    public static class Nettle extends NativeKeyPairGeneratorSpi {
+        public Nettle() {
+            initialize(256, new SecureRandom());
+        }
+
+        @Override
+        native boolean keysizeSupported(int keysize);
+
+        @Override
+        native boolean paramsSupported(AlgorithmParameterSpec params);
+
+        @Override
+        native KeyPair generate(int keysize, SecureRandom random);
+
+        @Override
+        KeyPair generate(AlgorithmParameterSpec params, SecureRandom random) {
+            if (params instanceof ECGenParameterSpec) {
+                // this might work? https://stackoverflow.com/questions/22646792/how-does-one-convert-a-public-ec-code-point-and-curve-name-into-a-publickey
+                try {
+                    AlgorithmParameters tmp = AlgorithmParameters.getInstance("EC");
+                    tmp.init(params);
+                    ECParameterSpec spec = tmp.getParameterSpec(ECParameterSpec.class);
+                    return generate(params, random, spec);
```

Comment:
I don't understand why an `ECGenParameterSpec` is converted to `ECParameterSpec` here, before passing it to the native module. Since the native `KeyPairGenerator` actually wants to use an `ECGenParameterSpec` because Nettle only supports curves by name and not explicit parameters.

Oh wait, now I get it. You need the `ECParameterSpec` to put it in the constructed keys, as they cannot take an `ECGenParameterSpec`. I would suggest not doing this kind of casting (because it actually likely uses the `SunEC` or `BouncyCastle` providers under the hood to convert the curves, when you do the `getInstance` call, any provider can provide its `AlgorithmParameters` class and the lookup from named params to explicit ones will be done with its set of curves, not ECTester's), but instead to use `EC_Store`, get the name from the named param spec, and look it up in `EC_Store` in the `secg` category (as all Nettle supported curves are there).

Reply:
Solved in d3ce67f